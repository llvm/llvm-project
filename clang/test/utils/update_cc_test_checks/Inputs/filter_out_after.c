// RUN: %clang_cc1 -triple=aarch64-unknown-linux-gnu -emit-llvm -o - %s | FileCheck -check-prefixes=CHECK %s

extern void bar(void);

void foo(int *p, int *q, double *t) {
  *p = 5;
  *q = 7;
  bar();
  *t = 1.3l;
}
