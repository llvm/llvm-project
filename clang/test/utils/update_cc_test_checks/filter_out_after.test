## Check that --filter_out_after works properly.
# RUN: cp -f %S/Inputs/filter_out_after.c %t.c && %update_cc_test_checks --version=5 --filter-out-after="call.*bar" %t.c
# RUN: diff -u %t.c %S/Inputs/filter_out_after.c.expected

## Check that running the script again does not change the result:
# RUN: %update_cc_test_checks --version=5 --filter-out-after="call.*bar" %t.c
# RUN: diff -u %t.c %S/Inputs/filter_out_after.c.expected

## Check that running the script again, without arguments, does not change the result:
# RUN: %update_cc_test_checks --version=5 %t.c
# RUN: diff -u %t.c %S/Inputs/filter_out_after.c.expected
