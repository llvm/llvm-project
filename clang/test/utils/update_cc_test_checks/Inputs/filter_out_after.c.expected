// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --filter-out-after "call.*bar" --version 5
// RUN: %clang_cc1 -triple=aarch64-unknown-linux-gnu -emit-llvm -o - %s | FileCheck -check-prefixes=CHECK %s

extern void bar(void);

// CHECK-LABEL: define dso_local void @foo(
// CHECK-SAME: ptr noundef [[P:%.*]], ptr noundef [[Q:%.*]], ptr noundef [[T:%.*]]) #[[ATTR0:[0-9]+]] {
// CHECK-NEXT:  [[ENTRY:.*:]]
// CHECK-NEXT:    [[P_ADDR:%.*]] = alloca ptr, align 8
// CHECK-NEXT:    [[Q_ADDR:%.*]] = alloca ptr, align 8
// CHECK-NEXT:    [[T_ADDR:%.*]] = alloca ptr, align 8
// CHECK-NEXT:    store ptr [[P]], ptr [[P_ADDR]], align 8
// CHECK-NEXT:    store ptr [[Q]], ptr [[Q_ADDR]], align 8
// CHECK-NEXT:    store ptr [[T]], ptr [[T_ADDR]], align 8
// CHECK-NEXT:    [[TMP0:%.*]] = load ptr, ptr [[P_ADDR]], align 8
// CHECK-NEXT:    store i32 5, ptr [[TMP0]], align 4
// CHECK-NEXT:    [[TMP1:%.*]] = load ptr, ptr [[Q_ADDR]], align 8
// CHECK-NEXT:    store i32 7, ptr [[TMP1]], align 4
// CHECK-NEXT:    call void @bar()
//
void foo(int *p, int *q, double *t) {
  *p = 5;
  *q = 7;
  bar();
  *t = 1.3l;
}
